### ISSUE TITLE
UnifiedMap: ConcurrentModificationException in GeoItemsLayer

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#15492]
Triggered by the error description in https://github.com/cgeo/cgeo/issues/15460#issuecomment-2010617620 I installed BalticCountries.map from OpenAndroMaps and tried it with UnifiedMap. Besides some effects described by @MagpieFourtyTwo I observed a crash in UnifiedMap's GeoItemsLayer:

```
03-20 22:06:34.927 16700 16700 W cgeo    : [main] UncaughtException
03-20 22:06:34.927 16700 16700 W cgeo    : java.util.ConcurrentModificationException
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1574)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashMap$EntryIterator.next(HashMap.java:1607)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashMap$EntryIterator.next(HashMap.java:1605)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashMap.putMapEntries(HashMap.java:509)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashMap.clone(HashMap.java:1467)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.util.HashSet.clone(HashSet.java:257)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer.lambda$new$5(GeoItemsLayer.java:94)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer.$r8$lambda$cXvET3P_pTcYCE6MIrnfetedcGw(GeoItemsLayer.java:0)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer$$ExternalSyntheticLambda5.onChanged(R8$$SyntheticClass:0)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at android.os.Handler.handleCallback(Handler.java:958)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at android.os.Looper.loopOnce(Looper.java:230)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at android.os.Looper.loop(Looper.java:319)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:8893)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
03-20 22:06:34.927 16700 16700 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

**Comment by eddiemuc** [comment]
PR #15511 might be a step into preventing such Concurrent-Modification-Exceptions for good in ViewModel LiveData classes

**Comment by moving-bits** [comment]
seems to be fixed by the related PR, at least haven't seen this error any longer, so let's close here for now

**Comment by eddiemuc** [comment]
> seems to be fixed by the related PR, at least haven't seen this error any longer, so let's close here for now

I am glad if the PR seems to help. But it can't possibly fix all related problems because I only applied the new mechanism to the Geocache hash set of the view model, and not the waypoint hash set. And I remember seeing at least one ccme while trying to access the waypoint hashset

**Comment by eddiemuc** [comment]
I had another CME today with live version (2024.06.02):

```
06-09 10:25:06.129 16454 16454 E AndroidRuntime: java.util.ConcurrentModificationException
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1574)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashMap$EntryIterator.next(HashMap.java:1607)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashMap$EntryIterator.next(HashMap.java:1605)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashMap.putMapEntries(HashMap.java:509)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashMap.clone(HashMap.java:1467)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.util.HashSet.clone(HashSet.java:257)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer.lambda$new$5(GeoItemsLayer.java:94)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer.$r8$lambda$cXvET3P_pTcYCE6MIrnfetedcGw(GeoItemsLayer.java:0)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer$$ExternalSyntheticLambda5.onChanged(R8$$SyntheticClass:0)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:233)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:344)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8212)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
06-09 10:25:06.129 16454 16454 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
```

This one is on the waypoints live data set. Thus the CME is not surprising since PR #15511 did remove the problem only for the geocache live data set. 
--> fix should be applied to waypoint live data set as well. I am reopening this issue

**Comment by eddiemuc** [comment]
Hopefully fixed with PR #15777
